> the CI seems crashed @stephanosio

@hakehuang It is a real problem caught by CI. See the raw logs:

```
2025-06-27T03:29:16.3211774Z In file included from <command-line>:
2025-06-27T03:29:16.3212535Z /__w/zephyr/zephyr/twister-out/mimxrt1180_evk_mimxrt1189_cm7/zephyr/samples/synchronization/sample.kernel.synchronization/zephyr/include/generated/zephyr/autoconf.h:5:44: error: missing binary operator before token "792000000"
2025-06-27T03:29:16.3213253Z     5 | #define CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC 792000000
2025-06-27T03:29:16.3213636Z       |                                            ^~~~~~~~~
2025-06-27T03:29:16.3214074Z /__w/zephyr/zephyr/include/zephyr/sys/time_units.h:82:49: note: in expansion of macro 'CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC'
2025-06-27T03:29:16.3214594Z    82 | #define sys_clock_hw_cycles_per_sec() (uint32_t)CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC
2025-06-27T03:29:16.3215022Z       |                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-06-27T03:29:16.3215435Z /__w/zephyr/zephyr/drivers/timer/cortex_m_systick.c:20:23: note: in expansion of macro 'sys_clock_hw_cycles_per_sec'
2025-06-27T03:29:16.3215906Z    20 | #define CYC_PER_TICK (sys_clock_hw_cycles_per_sec()     \
2025-06-27T03:29:16.3216256Z       |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-06-27T03:29:16.3216678Z /__w/zephyr/zephyr/drivers/timer/cortex_m_systick.c:27:20: note: in expansion of macro 'CYC_PER_TICK'
2025-06-27T03:29:16.3217098Z    27 | #if (COUNTER_MAX / CYC_PER_TICK) == 1
2025-06-27T03:29:16.3217392Z       |                    ^~~~~~~~~~~~
2025-06-27T03:29:16.3217681Z ninja: build stopped: subcommand failed.
```

--- a/drivers/timer/cortex_m_systick.c
+++ b/drivers/timer/cortex_m_systick.c
@@ -17,10 +17,22 @@
 #define COUNTER_MAX 0x00ffffff
 #define TIMER_STOPPED 0xff000000
 
-#define CYC_PER_TICK (sys_clock_hw_cycles_per_sec()	\
-		      / CONFIG_SYS_CLOCK_TICKS_PER_SEC)
-#define MAX_TICKS ((k_ticks_t)(COUNTER_MAX / CYC_PER_TICK) - 1)
+
+#if defined(CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME)
+extern unsigned int z_clock_hw_cycles_per_sec;
+#define CYC_PER_TICK (z_clock_hw_cycles_per_sec/CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+#else
+#define CYC_PER_TICK (CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC/CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+#endif
+
+/* add MAX_TICKS protection */
+#define _MAX_TICKS (int)((k_ticks_t)(COUNTER_MAX / CYC_PER_TICK) - 1)
+#define MAX_TICKS ((_MAX_TICKS > 0) ? _MAX_TICKS : 1)
 #define MAX_CYCLES (MAX_TICKS * CYC_PER_TICK)
+
+#if (COUNTER_MAX / CYC_PER_TICK) == 1
+#pragma message("tickless does nothing as CONFIG_SYS_CLOCK_TICKS_PER_SEC too low")
+#endif
 
 /* Minimum cycles in the future to try to program.  Note that this is
  * NOT simply "enough cycles to get the counter read and reprogrammed

